sql_generators/glean_usage/templates/baseline_clients_first_seen_v1.query.sql

```diff
@@ -76,17 +76,28 @@ _previous AS (
     AND first_seen_date < @submission_date
 )
 {% endif %}
-  --
-SELECT
-  IF(
-    _previous.client_id IS NULL
-    OR _previous.first_seen_date >= _current.first_seen_date,
-    _current,
+
+, _joined AS (
+  SELECT
+    IF(
+      _previous.client_id IS NULL
+      OR _previous.first_seen_date >= _current.first_seen_date,
+      _current,
+      _previous
+    ).*
+  FROM
+    _current
+  FULL JOIN
     _previous
-  ).*
-FROM
-  _current
-FULL JOIN
-  _previous
-USING
-  (client_id)
+  USING
+    (client_id)
+)
+
+-- added this as the result of bug#1788650
+SELECT
+  MIN(submission_date) AS submission_date,
+  MIN(first_seen_date) AS first_seen_date,
+  sample_id,
+  client_id
+FROM _joined
+GROUP BY sample_id, client_id
```

Comment:
this should also throw an error if `client_id` is not unique:

```suggestion
SELECT
  MIN(submission_date) AS submission_date,
  MIN(first_seen_date) AS first_seen_date,
  sample_id,
  client_id
FROM _joined
WHERE
  TRUE
QUALIFY
  IF(
    COUNT(*) OVER (PARTITION BY client_id) > 1,
    ERROR("duplicate client_id detected"),
    TRUE
  )
```

Reply:
this is neat, thanks @relud. You just became my go-to person for SQL questions 😄 

Could you explain the `WHERE TRUE` part?

Reply:
zetasql requires `QUALIFY` to be used in conjunction with `WHERE` or `HAVING`. BigQuery itself does not care, but we need zetasql to work because we use it to analyze the sql to find what tables are referenced. The `WHERE` clause doesn't need to do anything, it just needs to be present, so we use `WHERE TRUE` as a no-op to satisfy zetasql.

Reply:
thank you for the explanation! 🙇